### PR TITLE
Add vSphere CPI chart

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR:

- adds the vSphere CCM helm chart

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/cloud-provider-vsphere/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v2
+appVersion: "v1.19.0"
```
Comment (QuentinBisson) [inline helm/cloud-provider-vsphere/Chart.yaml:2]: I fixed it to 1.19 for now

Comment (QuentinBisson) [reply helm/cloud-provider-vsphere/Chart.yaml:2]: We will probably need to maintain multiple branches

File: helm/cloud-provider-vsphere/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+description: A Helm chart for cloud provider vsphere
+engine: gotpl
+home: https://github.com/giantswarm/cloud-provider-vsphere-app
+name: cloud-provider-vsphere
+version: [[ .Version ]]
```
Comment (MarcelMue) [inline helm/cloud-provider-vsphere/Chart.yaml:7]: Newline is missing here

File: helm/cloud-provider-vsphere/values.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  nodeSelector: {}
+  ## Allows for the default tolerations to be replaced with user-defined ones
+  tolerations: []
+
+service:
+  annotations: {}
+  # endpointPort: externally accessible port for UI and API
+  endpointPort: 43001
+  # targetPort: the internal port the UI and API are exposed on
+  targetPort: 43001
```
Comment (MarcelMue) [inline helm/cloud-provider-vsphere/values.yaml:63]: Same here!

File: helm/cloud-provider-vsphere/templates/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+metadata:
+  labels:
+    vsphere-cpi-infra: role-binding
+    {{- include "labels.common" . | nindent 4 }}
+  name: {{ include "resource.default.name" . }}-servicecatalog.k8s.io:apiserver-authentication-reader
+  namespace: {{ include "resource.default.namespace" . }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: extension-apiserver-authentication-reader
```
Comment (TheoBrigitte) [inline helm/cloud-provider-vsphere/templates/rbac.yaml:12]: This role is not defined in this chart.

Comment (QuentinBisson) [reply helm/cloud-provider-vsphere/templates/rbac.yaml:13]: This is one of the default kube role: kube-system    extension-apiserver-authentication-reader        2020-10-28T10:11:34Z

Comment (QuentinBisson) [reply helm/cloud-provider-vsphere/templates/rbac.yaml:13]: See https://kubernetes.io/docs/tasks/extend-kubernetes/setup-extension-api-server/

Comment (QuentinBisson) [reply helm/cloud-provider-vsphere/templates/rbac.yaml:15]: I added the comment in the rbac file

File: helm/cloud-provider-vsphere/Chart.yaml (hunk 2 above)
Comment (QuentinBisson) [reply helm/cloud-provider-vsphere/Chart.yaml]: Done

**Comment by cornelius-keller** [comment]
This is installed by the CAPI operator via cluster resource sets, so we might not need this. Current templates for clusterctl are using ClusterResourceSet to do this. We can do the same thing in our tooling  like kubectl-gs.

**Pull request closed without merging**